File: src/main/java/br/com/dusty/dkits/ability/Ability.java
package br.com.dusty.dkits.ability;

import br.com.dusty.dkits.gamer.Gamer;
import org.bukkit.event.Listener;

public class Ability implements Listener {
	
	public boolean isUsing(Gamer gamer) {
		return this.getClass().isAssignableFrom(gamer.getKit().getAbility().getClass());
	}
}

File: src/main/kotlin/br/com/dusty/dkits/ability/Ability.kt
package br.com.dusty.dkits.ability

import br.com.dusty.dkits.gamer.Gamer
import br.com.dusty.dkits.util.millisToPeriod
import br.com.dusty.dkits.util.text.Text
import org.bukkit.event.Listener

open class Ability: Listener {

	fun sendKitCooldownMessage(gamer: Gamer) {
		val period = gamer.kitCooldown.millisToPeriod().toInt()

		gamer.player.sendMessage(Text.negativePrefix().basic("Você deve ").negative("esperar").basic(" mais ").negative(period).basic((if (period == 1) " segundo" else " segundos") + " antes de usar essa ").negative(
				"habilidade").basic(" novamente!").toString())
	}

	fun hasAbility(gamer: Gamer) = gamer.kit.ability == this
}
